jina/optimizers/discovery.py

```diff
@@ -0,0 +1,128 @@
+import os
+import shutil
+from pathlib import Path
+
+from jina.executors import BaseExecutor
+from jina.logging import default_logger as logger
+from jina.helper import yaml
+
+
+def _read_file(filename):
+    with open(filename, "r") as file:
+        return file.readlines()
+
+
+def _extract_executor_files(flows):
```

Comment:
Not to be changed now, but this will make impossible to optimize executors user with ContainerPeas. 

In another iteration this must be addressed.

Reply:
In ContainerPeas the only way to parametrize executors is through the argument "uses_internal"

Reply:
Max is going to push the final changes for this file